core/io/resource_uid.cpp

```diff
@@ -346,10 +349,27 @@ String ResourceUID::get_path_from_cache(Ref<FileAccess> &p_cache_file, const Str
 	return String();
 }
 
-void ResourceUID::clear() {
-	cache_entries = 0;
-	unique_ids.clear();
-	changed = false;
+void ResourceUID::verify() {
+	MutexLock l(mutex);
+
+	Vector<ID> invalid_ids;
+	Ref<FileAccess> f = FileAccess::create(FileAccess::ACCESS_RESOURCES);
+
+	for (KeyValue<ID, Cache> &E : unique_ids) {
+		const String file = String::utf8(E.value.cs.ptr());
+		if (!f->file_exists(file) || ResourceLoader::get_resource_uid(file) != E.key) {
+			invalid_ids.push_back(E.key);
+		}
+	}
+
+	for (ID &id : invalid_ids) {
+		unique_ids.erase(id);
+	}
+
+	if (invalid_ids.size() > 0) {
```

Comment:
```suggestion
	if (!invalid_ids.is_empty()) {
```
You could also put the above loop under this condition.

Reply:
Thanks for the suggestion, done.